src/coreclr/jit/morph.cpp

```diff
@@ -11053,6 +11053,60 @@ GenTree* Compiler::fgMorphCommutative(GenTreeOp* tree)
     return op1;
 }
 
+//------------------------------------------------------------------------------
+// fgMorphOpCast : Try to simplify "(T)x op (T)y" to "(T)(x op y)".
+// https://github.com/dotnet/runtime/issues/13816
+//
+// Arguments:
+//       tree - node to fold
+//
+// return value:
+//       A folded GenTree* instance, or the same GenTree* if it couldn't be folded
+//
+
+GenTree* Compiler::fgMorphOpCast(GenTreeOp* tree)
```

Comment:
It's probably not worth adding a separate method or it needs a better name - this one is too generic and we already have `fgMorphCast`

Reply:
My vote would be for a better name. `fgMorphSmpOp` is 3K lines already.

Reply:
@SingleAccretion well, everything related to casts worth a separate file now e.g. `morph-casts.cpp` 🙂 (it's not a change request for this PR)

Reply:
If you're planning to move it to another file I think it's probably easiest for you if I keep it in a function and don't inline it into `fgMorphSmpOp`, right? I can rename it to something like `fgMorphCastedBitwiseOperation`, lmk if you have a better idea for a name!

Reply:
cdec019d463276d2546e47e51d6894e951f0a7b7